custom_components/ha_washdata/frontend.py

```
@@ -63,70 +75,229 @@ class LovelaceResourceItem(TypedDict, total=False):
+                os.stat(_resolve_asset(filename)).st_mtime_ns,
+            )
+        except OSError:
+            build_mtime = 0
+        mtime_part = str(max(src_mtime, trans_mtime, build_mtime))
     except OSError:
-        # Deterministic fallback when file is unavailable.
-        return "1"
+        mtime_part = "1"
 
+    raw = f"{manifest_version}:{mtime_part}"
+    return hashlib.sha1(raw.encode()).hexdigest()[:10]
```

Ruff (0.16.1)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)